src/http/web_socket.cr

```diff
@@ -68,18 +116,39 @@ class HTTP::WebSocket
       opcode = @opcode
     end
 
-    read = @io.read buffer[0, Math.min(@remaining, buffer.size)]
+    read = read_payload(buffer)
     @remaining -= read
+    PacketInfo.new(opcode, read.to_i, final? && @remaining == 0)
+  end
 
-    # Unmask payload, if needed
-    if masked?
-      read.times do |i|
-        buffer[i] ^= @mask[@mask_offset % 4]
-        @mask_offset += 1
-      end
+  private def write_header(size, opcode, masked, flags)
+    @io.write_byte(flags.value | opcode.value)
+
+    mask = masked ? MASK_BIT : 0_u8
+    if size <= 125
+      @io.write_byte(size.to_u8 | mask)
+    elsif size <= UInt16::MAX
+      @io.write_byte(126_u8 | mask)
+      size_u16 = size.to_u16
+      @io.write((pointerof(size_u16) as Pointer(UInt8[2])).value.reverse!.to_slice)
```

Comment:
Can't this be replaced by `@io.write_bytes(size.to_u16, IO::ByteFormat::NetworkEndian)`?